posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc

```diff
@@ -0,0 +1,411 @@
+* Resource allocation. As part of the application definition, you can tell ICCE how much CPU and Memory to allocate to any running instances. While the application is small, it is still a Java application that needs a period of startup before it can start serving requests. The Getting Started image will eventually start on the minimum values, but giving it slightly more will significantly improve startup and response time.
+* Listening port. Port 9080 should be used as the value for the listening port. Further information about HTTP handling within ICCE can be found https://cloud.ibm.com/docs/codeengine?topic=codeengine-application-workloads#considerationshttphandlingapp[here]
+
+Having created your application ensure that the application is not showing any errors such as `Missing Pull credentials` indicating that the image cannot be pulled to run. If you have any of these errors follow the steps in the ICCE documentation to resolve. Some errors may only occur when the application is invoked for the first time.
+
+=== Invoking your application on IBM Cloud Code Engine
+Having created the application, you can now invoke it within ICCE
+
+As part of creating your app you will have been able to obtain the URL from the test application or the CLI. if you did not obtain it via these means you can follow https://cloud.ibm.com/docs/codeengine?topic=codeengine-access-servICCE[accessing your app].
+
+If you make a request to `https://{ICCE_Application_URL}/` you should see the `Welcome to Open Liberty` page
+
+To call the application we use the same path as used for the application locally
+```
+curl https://${ICCE_Application_URL}/dev/system/properties
+```
+
+After a short time you should see a response with a JSON payload containing all the system properties.
+
+Note: All ICCE connections are HTTPS. so while `http` was used locally, the image is configured to support HTTPS without any changes.
+
+Congratulations you have now called your application on IBM Cloud Code Engine.
+
+While you are making the requests if you have the IBM Cloud Code Engine UI open on your applications Overview tab you will be able to see the active instances. you will need to be on the page when you start making the requests it creating the instance and then scaling down to zero when it stops receiving work
+
+image:/img/blog/ibm-code-engine-active-instances.png[IBM Cloud Code Engine Active Instances,width=70%,align="center"]
+
+=== Updating the application to use CloudEvents
+A common use case of serverless applications is to process events coming from non-http sources such as Kafka topics or Object Storage. Historically to consume these events the application had to make use of the Cloud Providers SDK, locking the application into that provider. This is where CloudEvents comes in, as it provides a cross-provider standard around which data can be transmitted and received by an application improving portability and reducing large dependencies.
```

Comment:
Please move this paragraph at line 130. and then add
`In this next section, we will demonstrate how to update your application to use CloudEvents.`